components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/topic/registry/RegistryTopicManager.java

```diff
@@ -393,16 +392,18 @@ private void addSubscriptions(String resourcePath,
                 }
             }
 
-            // add child subscriptions
+            // add child subscriptions only for resource collections
             if (withChildren) {
-                Collection childResources = (Collection) userRegistry.get(resourcePath);
-                for (String childResourcePath : childResources.getChildren()) {
-                    if ((!EventBrokerConstants.EB_CONF_WS_SUBSCRIPTION_COLLECTION_NAME
-                            .contains(childResourcePath)) &&
-                        (!EventBrokerConstants.EB_CONF_JMS_SUBSCRIPTION_COLLECTION_NAME
-                                .contains(childResourcePath))) {
-                        // i.e. this folder is a topic folder
-                        pathsQueue.add(childResourcePath);
+                if (userRegistry.get(resourcePath) instanceof Collection) {
+                    Collection childResources = (Collection) userRegistry.get(resourcePath);
```

Comment:
@AnujaLK you are calling userRegistry.get(resourcePath) method twice here. If registry caching is disabled, this will result in two DB calls. Is it possible to do the retrieval once, and re-use it later?